doc/Coding Guidelines.md

```diff
@@ -0,0 +1,107 @@
+# Vue Storefront Coding Guidelines - DRAFT
+
+Hello Extension / Theme / Core developer! This is just a short guide to align the code style among the core modules and all the extensions and thems.
+
+## General guidelines
+In general we're using kind of mix of: [Airbnb JavaScript Style Guide](https://github.com/airbnb/javascript) and [JavaScript Standard Style](https://github.com/feross/standard/blob/master/RULES.md#javascript-standard-style). There are a eslint and vue-eslint configured to enforce the style. Please do use the same [configs in Your modules](https://github.com/DivanteLtd/vue-storefront/blob/master/.eslintrc.js). 
+
+## ES2015
+We do preffer to use ES2015 constructions in the code - and there is a babel compiler set to allow us to do so. So use the ES modules instead of `require` and feel free to use other fancy structures.
```

Comment:
I'm not sure if we should recommend using ES6 for custom modules. Currently, we are using [some hacks](https://github.com/DivanteLtd/vue-storefront/blob/develop/src/extensions/index.js) to run Babel on code from `node_modules`, will be nice to get rid of it somewhere in the future.

Reply:
I think people will use it anyway as it is heavly used across whole project (even extensions are using es6 export  instead of module.exports)

Reply:
I think we rather should create a boilerplate / npm module for extension